assembly/src/v1/parsers/io_ops.rs

```diff
@@ -50,7 +50,218 @@ pub fn parse_readw(_span_ops: &mut Vec<Operation>, _op: &Token) -> Result<(), As
+/// on the stack, POP it off first. Then PAD 4 times to make space for the new elements, MOVUP5 the memory
+/// address to the top of the stack, and LOADW to load the word from memory into the padded spots
+///
+/// load: reads a word from memory and overwrites the top 4 elements of the stack. If an address was
+/// provided, PUSH the address, then LOADW to load a word from memory.
+///
+/// pop: pops the top 4 elements off the stack and saves them to memory. If an address was provided,
+/// PUSH the address first, STOREW to store the top 4 elements (below the address) at the specified memory address,
+/// then DROP 4 times to pop the elements off the stack
+///
+/// store: saves the top 4 elements of the stack to memory and leaves them on the stack. If an address was
+/// provided, PUSH the address first, then STOREW to save the top 4 elements to the specified memory location.
+pub fn parse_mem(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
+    match op.num_parts() {
+        0 | 1 => return Err(AssemblyError::missing_param(op)),
+        2 => (), // op is valid
+        3 => {
+            // parse the memory address provided by the assembly op
+            let address = parse_element_param(op, 2)?;
+
+            if op.parts()[1] == "push" {
+                // to save an instruction, handle mem.push.a differently from mem.push (below)
+                for _ in 0..4 {
+                    span_ops.push(Operation::Pad);
+                }
+                // push the memory address onto the stack for LoadW
+                span_ops.push(Operation::Push(address));
+                span_ops.push(Operation::LoadW);
+
+                return Ok(());
+            } else {
+                // after pushing the provided memory address onto the stack, all other 3-part mem op variants
+                // can be handled the same way as the 2-part op case where address is provided via the stack
+                span_ops.push(Operation::Push(address));
+            }
+        }
+        _ => return Err(AssemblyError::extra_param(op)),
+    }
+
+    // memory address is now assumed to be on top of the stack
+    match op.parts()[1] {
+        "store" => span_ops.push(Operation::StoreW),
+        "pop" => {
+            span_ops.push(Operation::StoreW);
+            for _ in 0..4 {
+                span_ops.push(Operation::Drop);
+            }
+        }
+        "push" => {
+            // only the 2-part op case where address is provided via the stack
+            for _ in 0..4 {
+                span_ops.push(Operation::Pad);
+            }
+            // move the memory address to the top of the stack
+            span_ops.push(Operation::MovUp4);
+            // push the address for LoadW
+            span_ops.push(Operation::LoadW);
+        }
+        "load" => span_ops.push(Operation::LoadW),
+        _ => return Err(AssemblyError::invalid_op(op)),
+    }
+
+    Ok(())
+}
+
```

Comment:
nit: maybe add separator lines here? I usually try to visually separate actual code from test code. Another option is to break out tests into a separate file.

Reply:
Added a separator line, since it seems to be idiomatic in Rust to keep unit tests in the same file.